code/modules/modular_computers/computers/item/laptop.dm

```diff
@@ -101,13 +102,19 @@
 
 /obj/item/modular_computer/laptop/proc/toggle_open(mob/living/user=null)
 	if(screen_on)
-		to_chat(user, span_notice("You close \the [src]."))
-		slowdown = initial(slowdown)
-		w_class = initial(w_class)
+		if(iscarbon(loc))
+			to_chat(user, span_notice("You close \the [src]."))
+			var/mob/living/carbon/wielder = loc
+			slowdown = initial(slowdown)
+			w_class = initial(w_class)
+			wielder.update_equipment_speed_mods()
```

Comment:
- `update_equipment_speed_mods` is mob level, the carbon cast doesn't make much sense. `ismob(loc) mob_loc.update_equipment_speed_mods()` works too
- You're only updating slowdown and weight class if held, do that regardless of if loc is a mob
- You also need to update `drag_slowdown` between on and off, an ddon't forget to do `if(isliving(loc)) mob_loc.update_pull_movespeed()`

Reply:
Done.